src/main/java/com/assignment1/parser/Validator.java

```diff
@@ -0,0 +1,157 @@
+package com.assignment1.parser;
+
+import com.assignment1.Type;
+import com.assignment1.exceptions.RuntimeExceptionCustom;
+import com.assignment1.item.ItemEntity;
+
+/**
+ * Class to implement Valid Interface for validation methods.
+ */
+@SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+public class Validator implements Valid {
```

Comment:
this implementation needs to be simplified

Reply:
Simplification is done.